src/stdlib/Localization.ts

```diff
@@ -0,0 +1,89 @@
+import * as fs from "fs";
+import * as path from "path";
+import { promisify } from "util";
+const readFile = promisify(fs.readFile);
+
+import { Interpreter } from "../interpreter";
+import { BrsString, Callable, ValueKind, StdlibArgument, RoDeviceInfo } from "../brsTypes";
+import { XmlDocument } from "xmldoc";
+
+/**
+ * Supported locales in RBI.
+ * @see https://developer.roku.com/docs/references/brightscript/interfaces/ifdeviceinfo.md#getcurrentlocale-as-string
+ */
+let locales = new Set<string>([
+    "en_US", // US English
+    "en_GB", // British English
+    "fr_CA", // Canadian French
+    "es_ES", // International Spanish
+    "de_DE", // German
+    "it_IT", // Italian
+    "pt_BR", // Brazilian Portuguese
+]);
+
+/** Source string to translated string */
+type Translations = Map<string, string>;
+
+/** File names to translation maps */
+let fileTranslations = new Map<string, Translations>();
+
+/**
+ * Parses and stores the actual translations of a given XML file.
+ * @param xmlNode The XmlDocument node that represents this translation file
+ * @param locale The locale for this translation file
+ */
+function parseTranslations(xmlNode: XmlDocument, locale: string) {
+    let translations: Translations = new Map();
+    let contextNode = xmlNode.childNamed("context");
+    contextNode?.childrenNamed("message").forEach((messageElement) => {
+        let source = messageElement.childNamed("source");
+        let translation = messageElement.childNamed("translation");
+        if (source && translation) {
+            translations.set(source.val, translation.val);
+        }
+    });
+
+    fileTranslations.set(locale, translations);
+}
+
+/**
+ * Finds and records all of the translation files that exist in the locale/ folder.
+ * @param rootDir The root package directory
+ */
+export async function loadTranslationFiles(interpreter: Interpreter, rootDir: string) {
+    locales.forEach(async (locale) => {
+        const filePath = path.join(rootDir, "locale", locale, "translations.ts");
+        if (fs.existsSync(filePath)) {
+            let xmlNode: XmlDocument;
+            try {
+                let contents = await readFile(filePath, "utf-8");
+                let xmlStr = contents.toString().replace(/\r?\n|\r/g, "");
+                xmlNode = new XmlDocument(xmlStr);
+            } catch (err) {
+                interpreter.stderr.write(`Error reading translations file ${filePath}: ${err}`);
+                return;
+            }
+
+            parseTranslations(xmlNode, locale);
+        }
+    });
+}
+
+export const Tr = new Callable("Tr", {
```

Comment:
Why isn't the `tr` function located where all the other global functions are?

Reply:
🤔 great question, i did not realize that! i can move it.

Reply:
Hmm actually what do you mean by this? In the `stdlib/` folder, there are other files, like `Math.ts`, `Json.ts`, `Print.ts`, etc. that define global functions. Is there another place that I should put `tr` instead of `stdlib/Localization.ts`?

Reply:
You're right, I thought we were putting our global utils in `stdlib/GlobalUtilities.ts` but they look categorized by function. I now think this is the correct place to put it :)